### PR TITLE
[Security] add mutex lock to the file loading in FileExternalAccountCredentials

### PR DESCRIPTION
**Pull request by rockspore**
I'm following the suggestions in #23503.

The FileExternalAccountCredentials is the only place I found that could potentially read the same file concurrently (if multiple requests are made simultaneously with the same call credentials).

However, I can't produce any errors without the lock so am not entirely certain whether this is a necessary fix.

### CODE REVIEW

**Review by gtcooke94 — APPROVED** [review]
This LGTM pending all the tests running

I'd also like to know if it's absolutely required at the point, so let's get Mark's eyes on it as well?

**Comment by rockspore** [comment]
@markdroth I just noticed you opened #35857. I think this shall wait till yours get merged.

File: src/core/lib/iomgr/load_file.h
Comment (markdroth) [inline src/core/lib/iomgr/load_file.h:30]: I am soon going to land #35857, which removes this API in favor of `grpc_core::LoadFile()`.  I suggest moving this comment to that function instead.

File: src/core/lib/security/credentials/external/file_external_account_credentials.h
Comment (markdroth) [inline src/core/lib/security/credentials/external/file_external_account_credentials.h:58]: Instead of using `gpr_mu` and `gpr_mu_lock()` / `gpr_mu_unlock()`, please use `grpc_core::Mutex` and `grpc_core::MutexLock`.

File: test/core/security/credentials_test.cc
Comment (markdroth) [inline test/core/security/credentials_test.cc:3080]: You're not actually spawning any new threads here, so everything is still happening single-threaded.  This is not actually going to exercise the synchronization you've added.

**Review by markdroth — COMMENTED** [review]
I added a few comments, but then I realized that this PR is not actually necessary in the first place, because the oauth credential fetching framework ensures that there is only one request outstanding at a time for a given credentials instance:

https://github.com/grpc/grpc/blob/521b2ea17ca1c2dadb83f4f76358f78bf190066f/src/core/lib/security/credentials/oauth2/oauth2_credentials.cc#L324

**Comment by rockspore** [comment]
Thanks for your review, @markdroth! Sorry I didn't catch that `start_fetch` boolean. And thanks for pointing out the flaw of the added test.

Nevertheless, I am closing the PR here.

**Pull request closed without merging**
